src/middleware.ts

```diff
@@ -1,12 +1,15 @@
 import {
-  State,
   GetState,
-  SetState,
-  StoreApi,
   PartialState,
+  SetState,
+  State,
   StateCreator,
+  StateStorage,
+  StoreApi,
 } from './vanilla'
 
+const isRunningInBrowser = () => typeof window !== 'undefined'
```

Comment:
I would like to avoid having this check in our middleware. Let's just leave it to users.